declarative-data-processing-for-dop/main.bal

```diff
@@ -0,0 +1,24 @@
+import ballerina/io;
+import ballerina/http;
```

Comment:
This sample is same as https://github.com/ballerina-guides/integration-samples/blob/main/working_with_data/working_with_data.bal

Can we simply use the one we already have?

Reply:
It's a little bit different. 
The following has group-by clause and a client invocation. 
If we are to update and use the same, we'll have to remove the client invocation. 
[working_with_data.bal](https://github.com/ballerina-guides/integration-samples/blob/main/working_with_data/working_with_data.bal) 

Or else, use exactly the same thing in https://github.com/ballerina-guides/integration-samples/blob/main/working_with_data/working_with_data.bal. wdyt?

Reply:
Will be reusing the existing sample. Did a small amendment to the existing sample adding a group-by clause in https://github.com/ballerina-guides/integration-samples/pull/28/commits/b4b6d45633bc0d0cf74322a6617d846a663ecd50

Hope that's fine
```ballerina
import ballerina/http;
import ballerina/io;

type Country record {
    string country;
    int population;
    string continent;
    int cases;
    int deaths;
};

// Prints the top 10 countries having the highest case-fatality ratio grouped by continent.
public function main() returns error? {
    http:Client diseaseEp = check new ("https://disease.sh/v3");
    Country[] countries = check diseaseEp->/covid\-19/countries;

    json summary =
        from var {country, continent, population, cases, deaths} in countries
            where population >= 100000 && deaths >= 100
            let decimal caseFatalityRatio = (<decimal>deaths / <decimal>cases * 100).round(4)
            let json countryInfo = {country, population, caseFatalityRatio}
            order by caseFatalityRatio descending
            limit 10
            group by continent
            order by avg(caseFatalityRatio)
            select {continent, countries: [countryInfo]};
    io:println(summary);
}

```